tests/spec/s-array.js

```diff
@@ -1596,4 +1596,376 @@ describe('Array', function () {
             expect(result).toEqual(['2', '3']);
         });
     });
+
+    describe('#sort()', function () {
+        describe('usage', function () {
+            it('requires a function or undefined as first argument', function () {
+                var actual = [];
+                expect(actual.sort()).toBe(actual);
+                expect(actual.sort(undefined)).toBe(actual);
+                expect(actual.sort(function () { return 0; })).toBe(actual);
+            });
+
+            it('requires a non-function or non-undefined to throw a `TypeError`', function () {
+                expect(function () { [].sort(null); }).toThrow(jasmine.any(TypeError));
```

Comment:
hmm, this might require `toThrow(new TypeError())`, and might require a message, but I'm also OK with not specifying in the test what kind of error it throws. just `toThrow` is enough, since no engine that throws throws a different kind of error.